metadata-ingestion/src/datahub/ingestion/extractor/schema_util.py

```diff
@@ -76,19 +110,13 @@ def avro_schema_to_mce_fields(avro_schema_string: str) -> List[SchemaField]:
     else:
         schema_parse_fn = avro.schema.Parse
 
-    parsed_schema: avro.schema.RecordSchema = schema_parse_fn(avro_schema_string)
+    parsed_schema: avro.schema.Schema = schema_parse_fn(avro_schema_string)
 
-    fields: List[SchemaField] = []
-    for parsed_field in parsed_schema.fields:
-        field = SchemaField(
-            fieldPath=parsed_field.name,
-            nativeDataType=str(parsed_field.type),
-            type=_get_column_type(parsed_field.type),
-            description=parsed_field.props.get("doc", None),
-            recursive=False,
-            nullable=_is_nullable(parsed_field),
-        )
+    if isinstance(parsed_schema, avro.schema.RecordSchema):
+        schema_convert_fn = _recordschema_to_mce_fields
+    else:
+        schema_convert_fn = _genericschema_to_mce_fields
```

Comment:
nit: is there a reason to create `schema_convert_fn` instead of simply calling the method and returning here?

Reply:
@hsheth2 Only that I wanted to follow the pattern above. Returning directly is indeed prettier.

Reply:
Ah makes sense 👍